src/client/dfuse/il/int_posix.c

```diff
@@ -1404,6 +1409,62 @@ dfuse_fclose(FILE *stream)
 	return __real_fclose(stream);
 }
 
+DFUSE_PUBLIC int
+dfuse___fxstat(int ver, int fd, struct stat *buf)
+{
+	struct fd_entry *entry = NULL;
+	int rc;
+
+	rc = vector_get(&fd_table, fd, &entry);
+	if (rc != 0)
+		goto do_real_fstat;
+
+	/* Turn of this feature if the kernel is doing metadata caching, in this case it's btter
+	 * to use the kernel cache and keep it up-to-date than query the severs each time.
+	 */
+	if (!entry->fd_fstat) {
+		vector_decref(&fd_table, entry);
+		goto do_real_fstat;
+	}
+
+	DFUSE_TRA_INFO(entry->fd_dfsoh, "fstat() %d", fd);
+
+	/* fstat needs to return both the device magic number and the inode
+	 * neither of which can change over time, but they're also not known
+	 * at this point.  For the first call to fstat do the real call
+	 * through the kernel, then save these two entries for next time.
+	 */
+	if (entry->fd_dev == 0) {
+		rc =  __real___fxstat(ver, fd, buf);
+		if (rc) {
+			vector_decref(&fd_table, entry);
+			return rc;
+		}
+		entry->fd_dev = buf->st_dev;
+		entry->fd_ino = buf->st_ino;
+		vector_decref(&fd_table, entry);
+		return 0;
+	}
+
+	rc = dfs_ostat(entry->fd_cont->ioc_dfs, entry->fd_dfsoh, buf);
+
+	DFUSE_TRA_INFO(entry->fd_dfsoh, "fstat() returned %d", rc);
```

Comment:
Minor: would this make more sense?
```suggestion
	rc = dfs_ostat(entry->fd_cont->ioc_dfs, entry->fd_dfsoh, buf);

	DFUSE_TRA_INFO(entry->fd_dfsoh, "dfs_ostat() returned %d", rc);
```